src/mono/mono/mini/mini-llvm.c

```diff
@@ -9683,7 +9683,49 @@ MONO_RESTORE_WARNING
 			break;
 		}
 #endif
-
+#if defined(TARGET_WASM)
+		case OP_WASM_SIMD_BITMASK: {
```

Comment:
This could use the OP_XOP_X_X opcode, there is not need for a separate one.

Reply:
I didn't find any suitable XOP for bitmask and swizzle intrinsics. The llvm has intrinsic functions for these too. Maybe I am looking at wrong places? Which X opcodes would fit for these?

Reply:
Looks like it doesn't exist, but swizzle is defined as:
def int_wasm_swizzle :
  Intrinsic<[llvm_v16i8_ty],
            [llvm_v16i8_ty, llvm_v16i8_ty],
            [IntrNoMem, IntrSpeculatable]>;

So it's just a binary operation on vectors, so it could use OP_XOP_X_X_X.

Reply:
I didn't find any suitable XOP. The wasm swizzle instruction (2 vectors shuffle, 2nd value acts as mask) is similar to sse shuffle. We are using sse special OPs for sse shuffle too (OP_SSE_SHUFPS, OP_SSE2_PSHUFD, OP_SSE2_SHUFPD). So I guess we don't have OP_XOP_... for shuffle.

IIRC, I also tried to use `LLVMBuildShuffleVector` at some point, which didn't yield the swizzle instruction selection for me, so I ended calling the intrinsic function. The llvm codegen tests also call it. The tests also test use of series of extractelement/insertelement, but that is not what we want.